replication-api-impl/src/test/java/com/connexta/replication/api/impl/data/FilterManagerImplTest.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.data;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.ion.replication.api.NotFoundException;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.impl.persistence.pojo.FilterPojo;
+import com.connexta.replication.api.impl.persistence.spring.FilterRepository;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FilterManagerImplTest {
+
+  private static final String ID = "id";
+
+  private static final String SITE_ID = "site_id";
+
+  private static final String FILTER = "filter";
+
+  private static final String NAME = "name";
+
+  private static final String DESCRIPTION = "description";
+
+  private static final byte PRIORITY = 2;
+
+  @Mock private FilterRepository filterRepository;
+
+  private FilterManagerImpl filterManager;
+
+  @Before
+  public void setup() {
+    filterManager = new FilterManagerImpl(filterRepository);
+  }
+
+  @Test
+  public void create() {
+    assertTrue(filterManager.create() instanceof FilterImpl);
+  }
+
+  @Test
+  public void get() {
+    when(filterRepository.findById(anyString())).thenReturn(Optional.of(makeDefaultPojo()));
+    Filter filter = filterManager.get(ID);
+    assertThat(filter.getId(), is(ID));
+    assertThat(filter.getSiteId(), is(SITE_ID));
+    assertThat(filter.getDescription().get(), is(DESCRIPTION));
+    assertThat(filter.getFilter(), is(FILTER));
+    assertThat(filter.getName(), is(NAME));
+    assertThat(filter.getPriority(), is(PRIORITY));
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void getFilterNotFound() {
+    when(filterRepository.findById(anyString())).thenReturn(Optional.empty());
+    filterManager.get("id");
+  }
+
```

Comment:
❗️ We should add a test case when it fails because the persitence layer throws a persistence exception